scripts/tests/test_validate_data.py

```
@@ -114,5 +120,360 @@ def test_non_dict_items_ignored(self):
+            share_id = new_short_url.generate_share_id(registry)
+            self.assertTrue(new_short_url.is_base62_id(share_id))
+            self.assertNotIn(share_id, registry)
+
+    def test_generate_share_id_prefers_two_chars(self):
+        registry = {"kn": {}, "lk": {}}
+        for _ in range(20):
+            self.assertEqual(len(new_short_url.generate_share_id(registry)), 2)
+
+    def test_generate_share_id_deterministic_with_rng(self):
+        registry = {"kn": {}, "lk": {}}
+        rng = random.Random(7)
```

Ruff (0.16.3)

[error] Standard pseudo-random generators are not suitable for cryptographic purposes

(S311)